system/jlib/jdebug.cpp

```diff
@@ -1966,6 +1977,73 @@ class CExtendedStats  // Disk network and cpu stats
 
         if (!getInterfaceName(ifname))
             ifname.set("eth0");
+
+#ifdef __linux__
+        // MCK - wish libblkid could do this ...
+        // Another way might also be to look for:
+        //   /sys/block/sd*
+        //   /sys/block/nvme*
+        // and match those with entries in /proc/diskstats
+        StringBuffer cmd("lsblk -o TYPE,MAJ:MIN --pairs");
+        Owned<IPipeProcess> pipe = createPipeProcess();
+        StringBuffer output;
+        if (pipe->run("list disks", cmd, nullptr, false, true))
+        {
+            Owned<ISimpleReadStream> pipeReader = pipe->getOutputStream();
+            readSimpleStream(output, *pipeReader);
+        }
+        int exitcode = pipe->wait();
+        if ( (exitcode == 0) && (output.length() > 0) )
+        {
```

Comment:
The below could be quite a bit simpler. I'd use StringArray and appendList to get into lines.
Then you can have do:
```
if (2 == sscanf(line, "TYPE=\"disk\" MAJ:MIN=\"%u:%u\"", &majnum, &minnum))
```
for all lines.

Reply:
ok, good idea, I will change

Reply:
Done